lib/Recur/RRuleIterator.php

```diff
@@ -1230,4 +1238,103 @@ protected function getMonths(): array
 
         return $recurrenceMonths;
     }
+
+    private function getNextIterationDateTime(\DateTimeInterface $dateTime, string $modify): \DateTimeInterface
+    {
+        $startTs = $dateTime->getTimestamp();
+        $initialDateTime = clone $dateTime;
+        $modifiedDateTime = $dateTime->modify($modify);
+
+        $this->detectAndSaveDstOffset(
+            $initialDateTime,
+            $modifiedDateTime,
+            $modify,
+        );
+
+        $modifiedDateTime = $this->revertPastOffset($modifiedDateTime);
+
+        return $modifiedDateTime;
+    }
+
+    private function detectAndSaveDstOffset(
+        \DateTimeInterface $initialDateTime,
+        \DateTimeInterface $modifiedDateTime,
+        string $modify
+    ) {
+        $dstTransitionLeap = $this->getDstTransitionLeap($initialDateTime, $modifiedDateTime, $modify);
+        if (!is_null($dstTransitionLeap)) {
+            $this->leapOffset[$this->counter] = ($this->leapOffset[$this->counter] ?? 0) + $dstTransitionLeap;
+        }
+
+    }
+
+    private function getDstTransitionLeap(
+        \DateTimeInterface $initialDateTime,
+        \DateTimeInterface $modifiedDateTime,
+        string $modify
+    ): ?int {
+        $modifiedDateInterval = $this->substractDates($modifiedDateTime, $initialDateTime);
+        $modifyInterval = \DateInterval::createFromDateString($modify);
+        $leap = $this->substractIntervals($modifiedDateInterval, $modifyInterval);
+
+        if ($this->isDstTransitionLeap($leap)) {
+            return 3_600 * $leap->h;
+        }
+
+        return null;
+    }
+
+    private function isDstTransitionLeap(\DateInterval $leap): bool
+    {
+        return $leap->y === 0 && $leap->m === 0 && $leap->d === 0 && $leap->h > 0 && $leap->i === 0 && $leap->s === 0;
+    }
+
+    private function revertPastOffset(\DateTimeInterface $dateTime): \DateTimeInterface
+    {
+        if (!is_numeric($this->counter)) {
+            return $dateTime;
+        }
+
+        $previousIterationLeapOffset = $this->leapOffset[(int) $this->counter - 1] ?? null;
+
+        if (!is_int($previousIterationLeapOffset)) {
+            return $dateTime;
+        }
+
+        $modifiedDateTime =  $dateTime->modify("- $previousIterationLeapOffset seconds");
+
+        unset($this->leapOffset[$this->counter - 1]);
+
+        return $modifiedDateTime;
+    }
+
```

Comment:
if you are anyway always resetting the previous offset, why keep a list of them? :thinking:

Reply:
Before my last commit, it was possible to call this method and "detectAndSaveDstOffset" multiple  times per iteration. It's probably not needed anymore indeed